### ISSUE TITLE
Handler set via `set_root` is ignored in `pfio.mkdir`

### ISSUE THREAD

**Issue by kmaehashi** [pfnet/pfio#142]
It seems the handler set via `set_root` is ignored in `pfio.mkdir`.

```py
>>> import pfio
>>> pfio.set_root(pfio.create_handler('hdfs'))  # or pfio.set_root('hdfs')
>>> pfio.mkdir('test/test')  # this is treated as Posix path
>>> pfio.open('test.txt')   # this is treated as HDFS path
```

(I was just playing around these APIs so this issue is low-priority (at least for me))

**Comment by belldandyxtq** [comment]
This is a known issue and we were trying to fix it in #99, but it ended up in redesign the API and the task became too huge. We didn't work on that after that, but please let us know if it became a problem, we then can allocate efforts to fix it.

**Comment by kuenishi** [comment]
Closing in favor of v2 API.
